Packages/Tracking/Core/Runtime/Plugins/LeapCSharp/Connection.cs

```diff
@@ -1024,6 +1024,56 @@ static public eLeapPolicyFlag FlagForPolicy(Controller.PolicyFlag singlePolicy)
             }
         }
 
+        /// <summary>
+        /// Temporarily makes a connection to determine if a Service is available.
+        /// Returns the result and closes the temporary connection upon completion.
+        /// </summary>
+        public static bool IsConnectionAvailable(string serverNamespace = "Leap Service")
+        {
+            LEAP_CONNECTION_CONFIG config = new LEAP_CONNECTION_CONFIG();
+            config.server_namespace = Marshal.StringToHGlobalAnsi(serverNamespace);
+            config.flags = 0;
+            config.size = (uint)Marshal.SizeOf(config);
+
+            IntPtr tempConnection;
+
+            eLeapRS result;
+
+            result = LeapC.CreateConnection(ref config, out tempConnection);
+
+            if (result != eLeapRS.eLeapRS_Success || tempConnection == IntPtr.Zero)
+            {
+                LeapC.CloseConnection(tempConnection);
+                return false;
+            }
+
+            result = LeapC.OpenConnection(tempConnection);
+
+            if (result != eLeapRS.eLeapRS_Success)
+            {
+                LeapC.CloseConnection(tempConnection);
+                return false;
+            }
+
+            LEAP_CONNECTION_MESSAGE _msg = new LEAP_CONNECTION_MESSAGE();
+            uint timeout = 150;
+            result = LeapC.PollConnection(tempConnection, timeout, ref _msg);
```

Comment:
What's the reason for polling once here?

Reply:
It is necessary to actually complete the connection (see [here](https://docs.ultraleap.com/tracking-api/group/group___functions.html#_CPPv418LeapOpenConnection15LEAP_CONNECTION))

Reply:
That's great. Thanks.